### PR TITLE
SPH-188-scenario-test-slling-contract

### PR DESCRIPTION
**Pull request by MaximeLemolt**

### CODE REVIEW

File: lib/PaygreenSdk/Payment/V3/Client.php [hunk 1]
```diff
@@ -68,6 +68,28 @@ public function authenticate()
...
 
+    /**
+     * @param string $username
+     * @param string $password
+     *
+     * @throws Exception
+     *
+     * @return ResponseInterface
+     */
+    public function authenticateWithCredentials($username, $password)
```
Comment (guillaumbertin) [inline lib/PaygreenSdk/Payment/V3/Client.php:79]: @MaximeLemolt pourquoi faire ca dans ce ticket ?

**Comment by guillaumbertin** [comment]
@MaximeLemolt 

@sellingContract  
Feature: Manage sellingContract  
  In order to manage sellingContract  
  Being authenticated  
  
The .env.behat file does not exist. This is probably unintentional.  
  Background: I have an authenticated Client # features/sellingContract.feature:6  
    Given A selling contract object          # FeatureContext::aSellingContractObject()  
    Given A ready to use Client              # FeatureContext::aReadyToUseClient()  
    And I authenticate in AM the Client      # FeatureContext::iAuthenticateInAMTheClient()  
      **Failed asserting that 400 matches expected 200.**  
  
  Scenario: Create a new selling contract              # features/sellingContract.feature:11  
    When I create a selling contract                   # FeatureContext::iCreateASellingContract()  
    Then I receive a 200 status code                   # FeatureContext::iReceiveAStatusCode()  
    And I receive a response with the selling contract # FeatureContext::iReceiveAResponseWithTheSellingContract()  
  
The .env.behat file does not exist. This is probably unintentional.  
  Scenario: Get all selling contracts                   # features/sellingContract.feature:16  
    And I authenticate in AM the Client                 # FeatureContext::iAuthenticateInAMTheClient()  
      **Failed asserting that 400 matches expected 200.**

File: features/bootstrap/AuthenticationDictionary.php [hunk 2]
```diff
@@ -22,6 +22,23 @@ public function iAuthenticateTheClient()
         $this->client->setBearer($bearer);
     }
 
+    /**
+     * @When /^I authenticate in AM the Client$/
+     */
+    public function iAuthenticateInAMTheClient()
```
Comment (guillaumbertin) [inline features/bootstrap/AuthenticationDictionary.php:28]: @MaximeLemolt idem, du coup tu as ajouté ces traitements dans deux branches ? (SPH-197)

File: lib/PaygreenSdk/Payment/V3/Client.php (hunk 1 above)
Comment (MaximeLemolt) [reply lib/PaygreenSdk/Payment/V3/Client.php:79]: @guillaumbertin j'en avais besoin pour les tests, ça aurait peut-être mérité d'être fait dans un ticket à part

File: features/bootstrap/AuthenticationDictionary.php (hunk 2 above)
Comment (MaximeLemolt) [reply features/bootstrap/AuthenticationDictionary.php:28]: @guillaumbertin Le SPH-197 est basé sur cette branche

**Pull request closed without merging**